plugins/generic/usageStats/UsageStatsOptoutBlockPlugin.inc.php

```diff
@@ -16,130 +16,91 @@
-	 * @param $parentPluginName string
+	 * @param $parentPlugin UsageStatsPlugin
 	 */
-	function __construct($parentPluginName) {
-		$this->_parentPluginName = $parentPluginName;
+	function __construct($parentPlugin) {
+		$this->_parentPlugin = $parentPlugin;
 		parent::__construct();
 	}
 
-
-	//
-	// Implement template methods from PKPPlugin.
-	//
 	/**
-	 * @see PKPPlugin::getHideManagement()
+	 * @copydoc PKPPlugin::getHideManagement()
 	 */
 	function getHideManagement() {
 		return true;
 	}
 
 	/**
-	 * @see PKPPlugin::getName()
+	 * @copydoc PKPPlugin::getName()
 	 */
 	function getName() {
 		return 'UsageStatsOptoutBlockPlugin';
 	}
 
 	/**
-	 * @see PKPPlugin::getDisplayName()
+	 * @copydoc PKPPlugin::getDisplayName()
 	 */
 	function getDisplayName() {
 		return __('plugins.reports.usageStats.optout.displayName');
 	}
 
 	/**
-	 * @see PKPPlugin::getDescription()
+	 * @copydoc PKPPlugin::getDescription()
 	 */
 	function getDescription() {
 		return __('plugins.generic.usageStats.optout.description');
 	}
 
 	/**
-	* @copydoc PKPPlugin::isSitePlugin()
-	*/
+	 * @copydoc PKPPlugin::isSitePlugin()
+	 */
 	function isSitePlugin() {
 		return false;
 	}
 
 	/**
-	 * @copydoc PKPPlugin::getTemplatePath()
+	 * @copydoc Plugin::getPluginPath()
 	 */
-	function getTemplatePath($inCore = false) {
-		$plugin =& $this->_getPlugin();
-		return $plugin->getTemplatePath(true); // FIXME Should we be forcing $inCore?
+	public function getPluginPath() {
+		return $this->_parentPlugin->getPluginPath();
 	}
 
 	/**
-	 * @copydoc BlockPlugin::getSeq()
+	 * @copydoc BlockPlugin::getEnabled()
 	 */
-	function getSeq($contextId = null) {
-		// Identify the position of the faceting block.
-		$seq = parent::getSeq($contextId);
-
-		// If nothing has been configured then show the privacy
-		// block after all other blocks in the context.
-		if (!is_numeric($seq)) $seq = 99;
-
-		return $seq;
+	function getEnabled($contextId = null) {
+		return $this->_parentPlugin->getEnabled($contextId);
 	}
 
-
-	//
-	// Implement template methods from LazyLoadPlugin
-	//
 	/**
-	 * @see LazyLoadPlugin::getEnabled()
+	 * @copydoc BlockPlugin::getContents()
 	 */
-	function getEnabled($contextId = null) {
-		$plugin =& $this->_getPlugin();
-		return $plugin->getEnabled($contextId);
+	function getContents($templateMgr, $request = null) {
+		return $templateMgr->fetch($this->getTemplateResourceName(true) . ':templates/' . $this->getBlockTemplateFilename());
 	}
 
-
-	//
-	// Implement template methods from BlockPlugin
-	//
 	/**
-	 * @see BlockPlugin::getBlockContext()
+	 * @copydoc BlockPlugin::getBlockContext()
 	 */
 	function getBlockContext() {
-		$blockContext = parent::getBlockContext();
-
-		// Place the block on the right by default.
-		if (!in_array($blockContext, $this->getSupportedContexts())) {
-			$blockContext = BLOCK_CONTEXT_SIDEBAR;
-		}
-
-		return $blockContext;
+		// if enabled, this block has to be in the sidebar
+		return BLOCK_CONTEXT_SIDEBAR;
```

Comment:
We may eventually have other block contexts (eg - footer). But when we do, we'll probably need to account for lots of cases like this. Just food for thought.

Reply:
OK, I will leave it as it is for now...